numba/targets/arrayobj.py

```diff
@@ -275,13 +275,23 @@ def getiter_array(context, builder, sig, args):
     return out
 
 
-def _getitem_array1d(context, builder, arrayty, array, idx, wraparound):
+def _getitem_array_int_only(context, builder, return_type, arrayty, array, idx,
+                            wraparound):
     """
     Look up and return an element from a 1D array.
     """
-    ptr = cgutils.get_item_pointer(context, builder, arrayty, array, [idx],
-                                   wraparound=wraparound)
-    return load_item(context, builder, arrayty, ptr)
+    dataptr, shapes, strides = cgutils.get_item_array_pointer(
+        context, builder, arrayty, array, [idx], wraparound=wraparound
+    )
+
+    if len(shapes) != 0:
```

Comment:
Elsewhere in `_getitem_array_generic` `if isinstance(return_type, types.Buffer):` is used instead.

Reply:
Changed to be the same, because really this is just an optimized form of `_getitem_array_generic`